windows/src/desktop/insthelp/main.pas

```diff
@@ -52,14 +52,20 @@ implementation
 procedure ShutdownKeyman;
 var
   hwnd: THandle;
+  Counter: Integer;
 const
   KMC_StopProduct = 1;  // from UfrmKeyman7Main.pas
 begin
   hwnd := FindWindow('TfrmKeyman7Main', nil);
   if hwnd <> 0 then
   begin
     PostMessage(hwnd, WM_USER_ParameterPass, KMC_StopProduct, 0);
-    Sleep(1000); // We'll wait 1 second to give Keyman Engine time to shut down
+    Counter := 0;
+    while (FindWindow('TfrmKeyman7Main', nil) <> 0) and (Counter < 10) do
+    begin
+      Sleep(1000); // We'll wait up to 10 seconds to give Keyman Engine time to shut down
+      Inc(Counter);
+    end;
```

Comment:
If we want to be really forthright:

```
if Counter == 0 then
begin
    // Warn user that a restart may be necessary (b/c shutdown unconfirmed)
end
```

But it's probably fine as-is anyway.

Reply:
They'll be warned anyway by Windows Installer, so don't think we need to do it here.